common/src/java/org/apache/hadoop/hive/conf/HiveConf.java

```diff
@@ -7002,6 +7018,16 @@ public void stripHiddenConfigurations(Configuration conf) {
     HiveConfUtil.stripConfigurations(conf, hiddenSet);
   }
 
+  /**
+   * Strips hidden config entries from configuration, but takes care of entries for execution engines.
+   */
+  public void stripHiddenConfigurationsForExecutionEngines(Configuration conf) {
+    Set<String> propsToBePropagatedToExecEngines = HiveConfUtil.getPropagateToExecutionEnginesList(conf);
+    Set<String> propsToBeStripped = new HashSet<String>(hiddenSet);
```

Comment:
```suggestion
    Set<String> propsToBeStripped = new HashSet<>(hiddenSet);
```
You should be able to use the diamond operator here.

Reply:
done